src/odemis/driver/pigcs.py

```diff
@@ -511,7 +511,7 @@ def recoverTimeout(self):
         try:
             for i in range(2):
                 self.busacc.flushInput()
-                resp = self.busacc.sendQueryCommand(self.address, "ERR?\n")
+                resp = self.busacc.sendQueryCommand(self.address, "ERR?\n").decode('latin1')
 
                 if isinstance(resp, list):
```

Comment:
This line is a sign that resp could sometimes be a `list`. In such case `decode()` will not work on it. So you need to either delay. Either just after this check, of maybe after the `re.match()`?